rpcserver.go

```diff
@@ -2702,6 +2702,17 @@ func abandonChanFromGraph(chanGraph *channeldb.ChannelGraph,
 func (r *rpcServer) abandonChan(chanPoint *wire.OutPoint,
 	bestHeight uint32) error {
 
+	// Before we remove the channel we mark the channel as confirmed so that
```

Comment:
So I included this part exactly because of the situation where we want to abandon a channel during the funding flow. My usecase in mind was, with this PR we will rebroadcast transactions which are either in Mempool or did not make it into the mempool because of the min_relay fee requirements. In either case if the noderunner decides to abandon a channel we must rely on his due diligence that this openchannel transaction will never confirm. If we do not remove the openchannel transaction form the Rebroadcaster he might run into the case where the funding flow leads to a channel not making it into the mempool because of the `min_relay_fee not met`. The user decides to abandon the channel because he has no way to bump the OpenChannelTx but the rebroadcaster keeps trying to broadcast this OpenChannelTx and we could run in a case where the user abandons the channel but the rebroadcaster succeeds in broadcasting it into the mempool (because `min_relay_fee` situation changes for example). So therefore we need to make sure this does not happen.

Reply:
Thanks for the detailed explanation! I think this is definitely needed, but could we do this in a new PR and focus on fixing the startup issue here so the PR can be merged more quickly?

Initially I had doubts because I thought there were certain logics implemented in `MarkAsConfirmed`. After checking it in neutrino I think it's fine as it just [deletes the request](https://github.com/lightninglabs/neutrino/blob/master/pushtx/broadcaster.go#L186). However, I think a proper way is to add a new method `CancelRequest`, thus each method has a more explicit description and responsibility, even tho atm the effects are the same.

Reply:
ok will do that in a separate PR.